packages/shared/src/hooks/vote/useVote.ts

```diff
@@ -40,11 +33,21 @@ const useVote = ({ onMutate, entity, variables }: UseVoteProps): UseVote => {
     {
       mutationKey,
       onMutate,
-      onSuccess: () => setIsVoting(false),
       onError: (err, _, rollback?: () => void) => rollback?.(),
     },
   );
 
+  const voteEntity: typeof onVoteEntity = useCallback(
+    (props) => {
+      if (isLoading) {
+        return Promise.resolve(null);
+      }
+
+      return onVoteEntity(props);
+    },
+    [isLoading, onVoteEntity],
+  );
```

Comment:
Just wondering to avoid anything like this in other places would it make sense to have any wrapper function for useMutation that would have this wrapper by default for mutateAsync? 

Of course changing that in all the places and making sure they work is a lot and not sure worth it, but we can consider in case we ever are faced with the same issue again

Reply:
I am still not in the case where I think we should wrap everything with this. We don't often do this as we normally prioritize optimistic UI. The root of us doing this was a deeper issue that happens if you spam the upvote.